plutus-core/plutus-ir/src/PlutusIR/TypeCheck/Internal.hs

```diff
@@ -313,7 +313,7 @@ checkTypeFromBinding recurs = \case
        checkConRes ty =
            -- We earlier checked that datacons' type is *-kinded (using checkKindBinding), but this is not enough:
            -- we must also check that its result type is EXACTLY `[[TypeCon tyarg1] ... tyargn]`
-           when (funResultType ty /= appliedTyCon) .
+           when (funResultType ty `eqIgnoringAnn` appliedTyCon) .
```

Comment:
I suggest instead passing `()` as the annotation on line 311, and then just calling `void` on the LHS here.

Reply:
You've also flipped the sense here, replacing `/=` with `==`. Could just change the `when` to `unless`.

Reply:
y thankfully the tests caught it.